test/integration/integration_test_fixtures.py

```diff
@@ -84,15 +90,22 @@ def setUp(self):
     """
     Performs sanity checks and starts up the server. Upon exit the server is ready to accept connections.
     """
-    assert (os.path.exists(self.nighthawk_test_server_path))
-    assert (os.path.exists(self.nighthawk_client_path))
-    test_id = os.environ.get('PYTEST_CURRENT_TEST').split(':')[-1].split(' ')[0].replace(
-        "[", "_").replace("]", "")
-    self.parameters["test_id"] = test_id
+    if os.getenv("NH_DOCKER_IMAGE", "") == "":
+      assert (os.path.exists(self.nighthawk_test_server_path))
+      assert (os.path.exists(self.nighthawk_client_path))
+
+    self.test_id = os.environ.get('PYTEST_CURRENT_TEST').split(':')[-1].split(' ')[0].replace(
```

Comment:
Ideally attributes on classes are all initialized (and documented if public) inside the __init__ method. Crating attributes "dynamically" in other methods makes it harder to find the complete list (API) of the class.

Can we initialize self.test_id to some useful value inside __init__? If nothing else, we can set it to the zero value of the same type (an empty string?).